sympy/stats/crv_types.py

```diff
@@ -2849,13 +2850,20 @@ class NormalDistribution(SingleContinuousDistribution):
 
     @staticmethod
     def check(mean, std):
-        _value_check(std > 0, "Standard deviation must be positive")
+        _value_check(std >= 0, "Standard deviation must be nonnegative.")
 
     def pdf(self, x):
+        # https://en.wikipedia.org/wiki/Normal_distribution#Zero-variance%20limit
+        if self.std == 0:
+            return DiracDelta(x - self.mean)
+
         return exp(-(x - self.mean)**2 / (2*self.std**2)) / (sqrt(2*pi)*self.std)
 
     def _cdf(self, x):
         mean, std = self.mean, self.std
+        if std == 0:
+            return Heaviside(x - mean)
+
         return erf(sqrt(2)*(-mean + x)/(2*std))/2 + S.Half
```

Comment:
Should this be `Piecewise` instead? What if `std` is a variable?

Reply:
I think `Piecewise` does make more sense and I was a little worried for backwards compatibility with return types. But after trying it, it seems to produce the same results. That is, it does not even return `Piecewise` in practice. 

It seems to have exactly the same functionality since `std == 0` Python's interpreter returns `True` or `False` whether inside an `if` statement or in a `Piecewise`. I'll change it as suggested since the source does look better.

Reply:
It looks like most of the distributions use `Piecewise`, so I think that's the way to go.